src/terminal/TelemetryService.hpp

```diff
@@ -2,6 +2,21 @@
 
 #include "Headers.hpp"
 
+#ifndef NO_TELEMETRY
```

Comment:
Move this into Headers.hpp so it's pre-compiled and in case other files want to use filesystem functions.

Reply:
also I think we can get rid of the NO_TELEMETRY wrapper and allow filesystem functions without telemetry.

Reply:
Will do.  

The original code in `Header.hpp` was `using namespace <x>;` and just using `boost` caused ambiguities in unrelated .cpp code, so moving the whole condition to `Telemetry.hpp` avoided that.  But I also realized that defining an unambiguous `fs` namespace and using that was a better approach anyway, so yeah, it'll be totally safe to move it back to `Header.`

I'm building and testing it now on a few weird old systems, and I'll push a commit when they finish.  The PocketBeagle takes about an hour and a half to build.  (That's not counting the time to build a CMake version new enough to work.). The M1 Mac takes about a minute.

Reply:
What I pushed works on modern Debian/Ubuntu, but fails on Raspberry Pi and Debian buster, so I'm testing better conditionalization now.  I'll push again once I've tested on a few systems.

Reply:
It required a bit of specific conditionalization so it only uses `boost::filesystem` when `std::filesystem` isn't there (but not on Mojave because `std::filesystem` is there but not usable).  Without that, it failed to link on systems with older Boost (latest Raspbian, for example).  But it now passes `et-test` everywhere I built it.

I recommend squashing all these commits, if that's an option.